File: csharp/Leet/LargestNumber.cs
// https://leetcode.com/problems/largest-number/
namespace Leet.LargestNumber
{
    public class Solution
    {
        public string LargestNumber(int[] nums)
        {
            var numStrings = new string[nums.Length];
            for (var i = 0; i < nums.Length; i++)
            {
                numStrings[i] = nums[i].ToString();
            }
            var numsString = numStrings.OrderDescending();

            var result = "";
            foreach (var s in numsString)
            {
                result += s;
            }
            return result;
        }
    }
}

File: python/today15/largest_number.py
# https://leetcode.com/problems/largest-number/
from typing import List
from functools import cmp_to_key

class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        def compare(x, y):
            return -1 if x+y > y+x else 1
        
        nums = list(map(str, nums))
        nums.sort(key=cmp_to_key(compare))
        
        if nums[0] == '0':
            return '0'

        return "".join(nums)
